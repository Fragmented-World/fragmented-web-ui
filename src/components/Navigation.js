import React from 'react'
import { Navbar, Nav, NavItem, NavDropdown, MenuItem } from 'react-bootstrap'

const Navigation = props => (
    <Navbar inverse>
        <Navbar.Header>
            <Navbar.Brand>
                <a href="/">Fragmented</a>
            </Navbar.Brand>
        </Navbar.Header>
        <Nav>
            <NavItem href="http://www.ryanyosua.me/decentralized-social-network/">About</NavItem>
            <NavItem href="http://www.ryanyosua.me/">Blog</NavItem>
            <NavDropdown title="Account" id="basic-nav-dropdown">
                <MenuItem>Sign up</MenuItem>
                <MenuItem>Log in</MenuItem>
                <MenuItem> Submissions</MenuItem>
            </NavDropdown>
        </Nav>
    </Navbar>
)

export default Navigation
