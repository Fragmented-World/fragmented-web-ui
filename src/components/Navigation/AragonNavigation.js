import React from 'react'
import { arrayOf, func, shape, string, bool, object } from 'prop-types'
import { AppBar } from '@aragon/ui'
import { NavigationBar } from '@aragon/ui'
//import ActionButton from 'components/ActionButton'
import { withRouter } from 'react-router-dom'

const AragonNavigation = ({
    isLoggedIn,
    notTippable,
    showRegisterAddressModal,
    setShowRegisterAddressModal,
    user,
    history,
    navigationItems,
    onBack
}) => (
    <AppBar title="Fragmented" onTitleClick={() => history.push('/')}>
        <NavigationBar items={navigationItems} onBack={onBack} />
    </AppBar>
)
/*
    <ActionButton label={text.Signup.title} />
    <ActionButton label={text.Login.title} />
    <ActionButton label={text.Submissions.title} />
*/

AragonNavigation.propTypes = {
    isLoggedIn: bool.isRequired,
    notTippable: bool.isRequired,
    showRegisterAddressModal: bool.isRequired,
    setShowRegisterAddressModal: func.isRequired,
    user: object,
    history: shape({
        push: func
    }),
    navigationItems: arrayOf(string),
    onBack: func
}

export default withRouter(AragonNavigation)
