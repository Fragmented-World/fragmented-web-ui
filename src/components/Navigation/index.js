import React from 'react'
import PropTypes from 'prop-types'
import { Navbar, Nav, NavItem, NavDropdown, MenuItem } from 'react-bootstrap'
import { LinkContainer } from 'react-router-bootstrap'
import text from 'util/text'
import RegisterPublicAddressModal from 'containers/RegisterPublicAddressModal'

const Navigation = ({ isLoggedIn, notTippable, showRegisterAddressModal, setShowRegisterAddressModal, user }) => (
    <Navbar inverse>
        <Navbar.Header>
            <Navbar.Brand>
                <a href="/newest">{text.app.name}</a>
            </Navbar.Brand>
            <Navbar.Toggle />
        </Navbar.Header>
        <Navbar.Collapse>
            <Nav>
                {isLoggedIn && <NavItem href="/create-post">{text.Posting.navigationLinkText}</NavItem>}
                <NavItem href="http://www.ryanyosua.me/decentralized-social-network/">{text.app.aboutLinkText}</NavItem>
                <NavItem href="http://www.ryanyosua.me/">{text.app.blogLinkText}</NavItem>
                <NavDropdown title="Account" id="basic-nav-dropdown">
                    {!isLoggedIn && (
                        <LinkContainer to="/signup">
                            <MenuItem>{text.Signup.title}</MenuItem>
                        </LinkContainer>
                    )}
                    {!isLoggedIn && (
                        <LinkContainer to="/login">
                            <MenuItem>{text.Login.title}</MenuItem>
                        </LinkContainer>
                    )}
                    {isLoggedIn && (
                        <LinkContainer to="/submissions">
                            <MenuItem>{text.Submissions.title}</MenuItem>
                        </LinkContainer>
                    )}
                    {isLoggedIn &&
                        notTippable && (
                            <MenuItem onSelect={() => setShowRegisterAddressModal(!showRegisterAddressModal)}>
                                {text.Tipping.unlinkedAccountNavHelp}
                            </MenuItem>
                        )}
                    {isLoggedIn && (
                        <LinkContainer to="/logout">
                            <MenuItem>{text.app.logout}</MenuItem>
                        </LinkContainer>
                    )}
                </NavDropdown>
            </Nav>
        </Navbar.Collapse>
        {user && (
            <RegisterPublicAddressModal
                show={showRegisterAddressModal && notTippable}
                onClose={() => setShowRegisterAddressModal(false)}
                user={user}
            />
        )}
    </Navbar>
)

Navigation.propTypes = {
    isLoggedIn: PropTypes.bool.isRequired,
    notTippable: PropTypes.bool.isRequired,
    showRegisterAddressModal: PropTypes.bool.isRequired,
    setShowRegisterAddressModal: PropTypes.func.isRequired,
    user: PropTypes.object
}

export default Navigation
