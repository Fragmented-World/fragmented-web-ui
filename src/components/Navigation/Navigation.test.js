import React from 'react'
import ReactDOM from 'react-dom'
import noop from 'lodash/noop'
import Navigation from 'components/Navigation'
import ShallowRenderer from 'react-test-renderer/shallow'

it('Navigation renders correctly', () => {
    const renderer = new ShallowRenderer()
    const result = renderer.render(
        <Navigation
            isLoggedIn={false}
            notTippable={false}
            showRegisterAddressModal={false}
            setShowRegisterAddressModal={noop}
        />
    )
    expect(result).toMatchSnapshot()
})
