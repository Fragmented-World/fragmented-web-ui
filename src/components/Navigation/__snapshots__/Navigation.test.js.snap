// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<Uncontrolled(Navbar)
  bsStyle="default"
  inverse={true}
>
  <NavbarHeader>
    <NavbarBrand>
      <a
        href="/newest"
      >
        Fragmented
      </a>
    </NavbarBrand>
    <NavbarToggle />
  </NavbarHeader>
  <NavbarCollapse>
    <Nav
      bsClass="nav"
      justified={false}
      pullLeft={false}
      pullRight={false}
      stacked={false}
    >
      <NavItem
        active={false}
        disabled={false}
        href="http://www.ryanyosua.me/decentralized-social-network/"
      >
        About
      </NavItem>
      <NavItem
        active={false}
        disabled={false}
        href="http://www.ryanyosua.me/"
      >
        Blog
      </NavItem>
      <NavDropdown
        id="basic-nav-dropdown"
        title="Account"
      >
        <LinkContainer
          activeClassName="active"
          exact={false}
          replace={false}
          strict={false}
          to="/signup"
        >
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
          >
            Signup
          </MenuItem>
        </LinkContainer>
        <LinkContainer
          activeClassName="active"
          exact={false}
          replace={false}
          strict={false}
          to="/login"
        >
          <MenuItem
            bsClass="dropdown"
            disabled={false}
            divider={false}
            header={false}
          >
            Login
          </MenuItem>
        </LinkContainer>
      </NavDropdown>
    </Nav>
  </NavbarCollapse>
</Uncontrolled(Navbar)>
`;
